### PR TITLE
taxonomy: Categories translation

### PR DESCRIPTION
**Pull request by teolemon**
### What
taxonomy: Categories translation
### ToDo
- [ ] Update failing test (because the Coffee category was translated)

https://github.com/openfoodfacts/openfoodfacts-server/blob/catcat/tests/integration/expected_test_results/api_cgi_suggest/categories-term-fr-cafe-accent.json
https://github.com/openfoodfacts/openfoodfacts-server/blob/catcat/tests/integration/expected_test_results/api_v3_taxonomy_suggestions/categories-string-fr-cafe-accent.json

```
atypus::TypeParser::ffi_type" used only once: possible typo at /usr/lib/x86_64-linux-gnu/perl-base/XSLoader.pm line 111.
tests/integration/add_update_to_redis.t ............ ok
Name "FFI::Platypus::keep" used only once: possible typo at /usr/lib/x86_64-linux-gnu/perl-base/XSLoader.pm line 111.
Name "FFI::Platypus::TypeParser::ffi_type" used only once: possible typo at /usr/lib/x86_64-linux-gnu/perl-base/XSLoader.pm line 111.
# Got a redirect to http://world.openfoodfacts.localhost/cgi/suggest.pl?tagtype=categories&term=Caf%C3%A9&lc=fr

#   Failed test 'categories-term-fr-cafe-accent'
#   at /opt/product-opener/lib/ProductOpener/Test.pm line 377.
# 
#     Structures begin differing at:
#          $got->[6] = 'Caf� au lait non sucr�'
#     $expected->[6] = 'Caf� au lait sucr�'
# 
# {
#   'expected_status_code' => 200,
#   'method' => 'GET',
#   'path' => "/cgi/suggest.pl?tagtype=categories&term=Caf\x{e9}&lc=fr",
#   'test_case' => 'categories-term-fr-cafe-accent'
# }
# [
#   "Caf\x{e9} Arabica",
#   "Caf\x{e9} Arabica moulu",
#   "Caf\x{e9} Robusta",
#   "Caf\x{e9} Robusta moulu",
#   "Caf\x{e9} au lait en poudre soluble",
#   "Caf\x{e9} au lait instantan\x{e9} non sucr\x{e9}",
#   "Caf\x{e9} au lait non sucr\x{e9}",
#   "Caf\x{e9} au lait sucr\x{e9}",
#   "Caf\x{e9} br\x{e9}silien",
#   "Caf\x{e9} cr\x{e8}me instantan\x{e9} non sucr\x{e9}",
#   "Caf\x{e9} cr\x{e8}me non sucr\x{e9}",
#   "Caf\x{e9} d'Inde",
#   "Caf\x{e9} d'Indon\x{e9}sie",
#   "Caf\x{e9} d'Ouganda",
#   "Caf\x{e9} d'\x{c9}thiopie",
#   "Caf\x{e9} de Colombie",
#   "Caf\x{e9} de C\x{f4}te d'Ivoire",
#   "Caf\x{e9} de Madagascar",
#   "Caf\x{e9} de Tanzanie",
#   "Caf\x{e9} des Philippines",
#   "Caf\x{e9} du Costa Rica",
#   "Caf\x{e9} du Guatemala",
#   "Caf\x{e9} du Honduras",
#   "Caf\x{e9} du Laos",
#   "Caf\x{e9} du Mexique"
# ]
# 
# Got a redirect to http://world.openfoodfacts.localhost/cgi/suggest.pl?tagtype=categories&string=Caf%C3%A9&lc=fr

#   Failed test 'categories-string-fr-cafe-accent'
#   at /opt/product-opener/lib/ProductOpener/Test.pm line 377.
# 
#     Structures begin differing at:
#          $got->[6] = 'Caf� au lait non sucr�'
#     $expected->[6] = 'Caf� au lait sucr�'
# 
# {
#   'expected_status_code' => 200,
#   'method' => 'GET',
#   'path' => "/cgi/suggest.pl?tagtype=categories&string=Caf\x{e9}&lc=fr",
#   'test_case' => 'categories-string-fr-cafe-accent'
# }
# [
#   "Caf\x{e9} Arabica",
#   "Caf\x{e9} Arabica moulu",
#   "Caf\x{e9} Robusta",
#   "Caf\x{e9} Robusta moulu",
#   "Caf\x{e9} au lait en poudre soluble",
#   "Caf\x{e9} au lait instantan\x{e9} non sucr\x{e9}",
#   "Caf\x{e9} au lait non sucr\x{e9}",
#   "Caf\x{e9} au lait sucr\x{e9}",
#   "Caf\x{e9} br\x{e9}silien",
#   "Caf\x{e9} cr\x{e8}me instantan\x{e9} non sucr\x{e9}",
#   "Caf\x{e9} cr\x{e8}me non sucr\x{e9}",
#   "Caf\x{e9} d'Inde",
#   "Caf\x{e9} d'Indon\x{e9}sie",
#   "Caf\x{e9} d'Ouganda",
#   "Caf\x{e9} d'\x{c9}thiopie",
#   "Caf\x{e9} de Colombie",
#   "Caf\x{e9} de C\x{f4}te d'Ivoire",
#   "Caf\x{e9} de Madagascar",
#   "Caf\x{e9} de Tanzanie",
#   "Caf\x{e9} des Philippines",
#   "Caf\x{e9} du Costa Rica",
#   "Caf\x{e9} du Guatemala",
#   "Caf\x{e9} du Honduras",
#   "Caf\x{e9} du Laos",
#   "Caf\x{e9} du Mexique"
# ]
# 
# Got a redirect to http://world.openfoodfacts.localhost/cgi/suggest.pl?tagtype=categories&string=jus%20de%20fruits%20au%20lait&lc=fr
# Looks like you failed 2 tests of 36.
```

### CODE REVIEW

**Comment by teolemon** [comment]
ERROR - hu:világos-sörök already is a synonym of hu:lágerek for entry en:lagers (categories) - hu:világos-sörök cannot be mapped to entry en:pale-lagers / hu:világos-sörök
ERROR - it:birre-chiare already is associated to en:lagers (categories) - it:birre-chiare cannot be mapped to entry en:pale-lagers
ERROR - ja:ラガービール already is associated to en:lagers (categories) - ja:ラガービール cannot be mapped to entry en:pale-lagers
ERROR - de:garnelen already is associated to de:garnelen (categories) - de:garnelen cannot be mapped to entry en:prawns

File: taxonomies/categories.txt [hunk 1]
```diff
@@ -2053,6 +2071,18 @@ description:en:Hefeweizen is a lager that originated from Bavaria. It is unfilte
...
+es:Cervezas claras
+fi:Vaaleat lagerit
+fr:Lagers blondes
+he:לאגרים בהירים
+hu:Világos sörök
+it:Birre chiare
+ja:ラガービール
+lt:Šviesūs lageriai
+nl:Lichte lagers
+pl:Jasne lagerowe
```
Comment (teolemon) [inline taxonomies/categories.txt:2085]:
```suggestion
```

File: taxonomies/categories.txt [hunk 2]
```diff
@@ -88810,8 +89068,12 @@ ja:ホンホッコクアカエビ
 nl:Pandalis Borealis
 
 <en:Shrimps
+de:Garnelen
```
Comment (teolemon) [inline taxonomies/categories.txt:89071]:
```suggestion
```

File: taxonomies/categories.txt [hunk 3]
```diff
@@ -5980,26 +6094,29 @@ ciqual_food_name:fr:Cola, sucré, sans caféine
...
 de:Pflanzliche Getränke
 es:Bebidas de origen vegetal
-fi:kasvipohjaiset juomat
+fi:Kasvipohjaiset juomat
 fr:Boissons à base de végétaux
 he:משקאות מבוססי צמחים
 hu:Növényi alapú italok
 it:Bevande a base di piante
-lt:Augalinės kilmės gėrimai, Augaliniai gėrimai
+lt:Augalinės kilmės gėrimai
```
Comment (teolemon) [inline taxonomies/categories.txt:6107]:
```suggestion
lt:Augalinės kilmės gėrimai, Augaliniai gėrimai
```

File: taxonomies/categories.txt [hunk 4]
```diff
@@ -5980,26 +6094,29 @@ ciqual_food_name:fr:Cola, sucré, sans caféine
...
 nl:Plantaardige dranken
 nl_be:Plantaardige dranken
-pl:Napoje na bazie roślin, Napoje roślinne
+pl:Napoje na bazie roślin
```
Comment (teolemon) [inline taxonomies/categories.txt:6110]:
```suggestion
pl:Napoje na bazie roślin, Napoje roślinne
```

File: taxonomies/categories.txt [hunk 5]
```diff
@@ -5980,26 +6094,29 @@ ciqual_food_name:fr:Cola, sucré, sans caféine
...
 pt:Bebidas à base de plantas
 ro:Băuturi din plante
 ru:Напитки из растений
 sv:Växtbaserade drycker
 th:เครื่องดื่มจากพืช
-tr:Bitkisel içecekler, Bitki kaynaklı içecekler, Bitki içecekleri
+tr:Bitkisel içecekler
```
Comment (teolemon) [inline taxonomies/categories.txt:6116]:
```suggestion
tr:Bitkisel içecekler, Bitki kaynaklı içecekler, Bitki içecekleri
```

File: taxonomies/categories.txt [hunk 6]
```diff
@@ -10340,8 +10527,9 @@ nl:Watermeloensiropen, Watermeloensiroop
 pl:Syropy arbuzowe
 
 <en:Flavoured syrups
-fr:Sirops de pamplemousse, Sirops de pamplemousses
 de:Grapefruitsirups
+en:Grapefruit syrups
+fr:Sirops de pamplemousse, Sirops de pamplemousses
```
Comment (teolemon) [inline taxonomies/categories.txt:10532]:
```suggestion
en:Grapefruit syrups
de:Grapefruitsirups
fr:Sirops de pamplemousse, Sirops de pamplemousses
```

File: taxonomies/categories.txt [hunk 7]
```diff
@@ -35969,8 +36201,13 @@ en:White chocolate bar with raisins
 lt:Balti šokoladai su razinomis
 
 <en:White chocolate bar with dried fruits
-fr:Chocolat blanc en tablette aux praliné
+de:Weiße Schokolade mit Praliné
 en:White chocolate bar with praline
```
Comment (teolemon) [inline taxonomies/categories.txt:36205]:
```suggestion
en:White chocolate bar with praline
de:Weiße Schokolade mit Praliné
```

**Comment by teolemon** [comment]
ERROR - hu:világos-sörök already is a synonym of hu:lágerek for entry en:lagers (categories) - hu:világos-sörök cannot be mapped to entry en:pale-lagers / hu:világos-sörök
ERROR - it:birre-chiare already is associated to en:lagers (categories) - it:birre-chiare cannot be mapped to entry en:pale-lagers
ERROR - ja:ラガービール already is associated to en:lagers (categories) - ja:ラガービール cannot be mapped to entry en:pale-lagers
ERROR - bg:растителни-напитки already is a synonym of bg:растителни-напитки for entry en:plant-based-beverages (categories) - bg:растителни-напитки cannot be mapped to entry en:plant-based-milk-alternatives / bg:растителни-млека

**Comment by teolemon** [comment]
# error: Error - code not a number with 8 or more digits {code => 4083637,i => 25,imported_product_ref => {abbreviated_product_name_en => "",abbreviated_product_name_fr => "",allergens => "Lait",allergens_tags => "en:milk",brand_owner => "",brands => "R\x{e9}gilait",brands_tags => "regilait",calcium_unit => "",calcium_value => "",carbohydrates_unit => "g",carbohydrates_value => 10,categories => "Produits laitiers, Laits concentr\x{e9}s, Laits concentr\x{e9}s non sucr\x{e9}s, Laits entiers concentr\x{e9}s",categories_tags => "en:dairies,en:evaporated-milks,fr:laits-concentres-non-sucres,fr:laits-entiers-concentres",cholesterol_unit => "",cholesterol_value => "",code => 4083637,conservation_conditions_fr => "",countries => "France",countries_tags => "en:france",customer_service_fr => "",data_sources => "App - yuka, Apps",emb_codes => "DE RP-25 EC",emb_codes_tags => "de-rp-25-ec","energy-kcal_unit" => "kcal","energy-kcal_value" => 132,"energy-kj_unit" => "","energy-kj_value" => "",energy_unit => "kcal",energy_value => 132,fat_unit => "g",fat_value => "7,5",fiber_unit => "",fiber_value => "","fruits-vegetables-nuts-estimate_unit" => "","fruits-vegetables-nuts-estimate_value" => "",generic_name_en => "",generic_name_fr => "",generic_name_pt => "",ingredients_text_en => "",ingredients_text_es => "",ingredients_text_fr => "lait entier (origine UE), stabilisants : phosphates de sodium, carragh\x{e9}nanes",ingredients_text_pt => "",iron_unit => "",iron_value => "",labels => "Fabriqu\x{e9} en Allemagne",labels_tags => "en:made-in-germany",lc => "fr",link => "",manganese_unit => "",manganese_value => "",manufacturing_places => "",manufacturing_places_tags => "","monounsaturated-fat_unit" => "","monounsaturated-fat_value" => "",nutrition_data_per => "100g",nutrition_data_prepared_per => "100g",obsolete => 0,obsolete_since_date => "","off:ecoscore_data.agribalyse.code" => 19026,"off:ecoscore_grade" => "b","off:ecoscore_score" => 64,"off:nutriscore_grade" => "c","off:nutriscore_score" => 5,origins => "",origins_tags => "",packaging => "",packaging_1_material => "",packaging_1_number_of_units => "",packaging_1_quantity_per_unit => "",packaging_1_shape => "",packaging_1_weight_measured => "",packaging_1_weight_specified => "",packaging_2_material => "",packaging_2_number_of_units => "",packaging_2_quantity_per_unit => "",packaging_2_shape => "",packaging_2_weight_measured => "",packaging_2_weight_specified => "",packaging_tags => "","polyunsaturated-fat_unit" => "","polyunsaturated-fat_value" => "",potassium_unit => "",potassium_value => "",preparation_fr => "",producer_product_id => "",producer_version_id => "",product_name_de => "",product_name_en => "",product_name_es => "",product_name_fr => "Lait concentr\x{e9} non sucr\x{e9} entier",product_name_pt => "",proteins_unit => "g",proteins_value => "6,1",quantity => "410 g",salt_unit => "g",salt_value => "0,25","saturated-fat_unit" => "g","saturated-fat_value" => 5,serving_size => "",sodium_unit => "g",sodium_value => "0,1","sources_fields:org-database-usda:available_date" => "","sources_fields:org-database-usda:fdc_category" => "","sources_fields:org-database-usda:fdc_data_source" => "","sources_fields:org-database-usda:fdc_id" => "","sources_fields:org-database-usda:modified_date" => "","sources_fields:org-database-usda:publication_date" => "","sources_fields:org-gs1:gln" => "","sources_fields:org-gs1:gpcCategoryCode" => "","sources_fields

**Comment by teolemon** [comment]
@stephanegigandet I'm not sure what's going wrong on the test

**Comment by teolemon** [comment]
atypus::TypeParser::ffi_type" used only once: possible typo at /usr/lib/x86_64-linux-gnu/perl-base/XSLoader.pm line 111.
tests/integration/add_update_to_redis.t ............ ok
Name "FFI::Platypus::keep" used only once: possible typo at /usr/lib/x86_64-linux-gnu/perl-base/XSLoader.pm line 111.
Name "FFI::Platypus::TypeParser::ffi_type" used only once: possible typo at /usr/lib/x86_64-linux-gnu/perl-base/XSLoader.pm line 111.
# Got a redirect to http://world.openfoodfacts.localhost/cgi/suggest.pl?tagtype=categories&term=Caf%C3%A9&lc=fr

#   Failed test 'categories-term-fr-cafe-accent'
#   at /opt/product-opener/lib/ProductOpener/Test.pm line 377.
# 
#     Structures begin differing at:
#          $got->[6] = 'Caf� au lait non sucr�'
#     $expected->[6] = 'Caf� au lait sucr�'
# 
# {
#   'expected_status_code' => 200,
#   'method' => 'GET',
#   'path' => "/cgi/suggest.pl?tagtype=categories&term=Caf\x{e9}&lc=fr",
#   'test_case' => 'categories-term-fr-cafe-accent'
# }
# [
#   "Caf\x{e9} Arabica",
#   "Caf\x{e9} Arabica moulu",
#   "Caf\x{e9} Robusta",
#   "Caf\x{e9} Robusta moulu",
#   "Caf\x{e9} au lait en poudre soluble",
#   "Caf\x{e9} au lait instantan\x{e9} non sucr\x{e9}",
#   "Caf\x{e9} au lait non sucr\x{e9}",
#   "Caf\x{e9} au lait sucr\x{e9}",
#   "Caf\x{e9} br\x{e9}silien",
#   "Caf\x{e9} cr\x{e8}me instantan\x{e9} non sucr\x{e9}",
#   "Caf\x{e9} cr\x{e8}me non sucr\x{e9}",
#   "Caf\x{e9} d'Inde",
#   "Caf\x{e9} d'Indon\x{e9}sie",
#   "Caf\x{e9} d'Ouganda",
#   "Caf\x{e9} d'\x{c9}thiopie",
#   "Caf\x{e9} de Colombie",
#   "Caf\x{e9} de C\x{f4}te d'Ivoire",
#   "Caf\x{e9} de Madagascar",
#   "Caf\x{e9} de Tanzanie",
#   "Caf\x{e9} des Philippines",
#   "Caf\x{e9} du Costa Rica",
#   "Caf\x{e9} du Guatemala",
#   "Caf\x{e9} du Honduras",
#   "Caf\x{e9} du Laos",
#   "Caf\x{e9} du Mexique"
# ]
# 
# Got a redirect to http://world.openfoodfacts.localhost/cgi/suggest.pl?tagtype=categories&string=Caf%C3%A9&lc=fr

#   Failed test 'categories-string-fr-cafe-accent'
#   at /opt/product-opener/lib/ProductOpener/Test.pm line 377.
# 
#     Structures begin differing at:
#          $got->[6] = 'Caf� au lait non sucr�'
#     $expected->[6] = 'Caf� au lait sucr�'
# 
# {
#   'expected_status_code' => 200,
#   'method' => 'GET',
#   'path' => "/cgi/suggest.pl?tagtype=categories&string=Caf\x{e9}&lc=fr",
#   'test_case' => 'categories-string-fr-cafe-accent'
# }
# [
#   "Caf\x{e9} Arabica",
#   "Caf\x{e9} Arabica moulu",
#   "Caf\x{e9} Robusta",
#   "Caf\x{e9} Robusta moulu",
#   "Caf\x{e9} au lait en poudre soluble",
#   "Caf\x{e9} au lait instantan\x{e9} non sucr\x{e9}",
#   "Caf\x{e9} au lait non sucr\x{e9}",
#   "Caf\x{e9} au lait sucr\x{e9}",
#   "Caf\x{e9} br\x{e9}silien",
#   "Caf\x{e9} cr\x{e8}me instantan\x{e9} non sucr\x{e9}",
#   "Caf\x{e9} cr\x{e8}me non sucr\x{e9}",
#   "Caf\x{e9} d'Inde",
#   "Caf\x{e9} d'Indon\x{e9}sie",
#   "Caf\x{e9} d'Ouganda",
#   "Caf\x{e9} d'\x{c9}thiopie",
#   "Caf\x{e9} de Colombie",
#   "Caf\x{e9} de C\x{f4}te d'Ivoire",
#   "Caf\x{e9} de Madagascar",
#   "Caf\x{e9} de Tanzanie",
#   "Caf\x{e9} des Philippines",
#   "Caf\x{e9} du Costa Rica",
#   "Caf\x{e9} du Guatemala",
#   "Caf\x{e9} du Honduras",
#   "Caf\x{e9} du Laos",
#   "Caf\x{e9} du Mexique"
# ]
# 
# Got a redirect to http://world.openfoodfacts.localhost/cgi/suggest.pl?tagtype=categories&string=jus%20de%20fruits%20au%20lait&lc=fr
# Looks like you failed 2 tests of 36.

**Comment by teolemon** [comment]
https://github.com/openfoodfacts/openfoodfacts-server/blob/catcat/tests/integration/expected_test_results/api_cgi_suggest/categories-term-fr-cafe-accent.json
https://github.com/openfoodfacts/openfoodfacts-server/blob/catcat/tests/integration/expected_test_results/api_v3_taxonomy_suggestions/categories-string-fr-cafe-accent.json

**Review by alexgarel — APPROVED** [review]
Thanks !

**Pull request merged**